src/ffi/spatialmath/vector3.rs

```diff
@@ -0,0 +1,121 @@
+use ffi_helpers::null_pointer_check;
+use libc::c_double;
+
+use crate::spatialmath::vector3::Vector3;
+
+#[no_mangle]
+pub extern "C" fn new_vector3(x: f64, y: f64, z: f64) -> *mut Vector3 {
+    Vector3::new(x, y, z).to_raw_pointer()
+}
+
+#[no_mangle]
+pub extern "C" fn free_vector_memory(ptr: *mut Vector3) {
+    if ptr.is_null() {
+        return;
+    }
+    unsafe {
+        Box::from_raw(ptr);
```

Comment:
What is this line supposed to be doing? It looks to me like it's creating a box but not doing anything with it, which would only free the ptr in a pretty roundabout way. Can this be more explicit? Note also that clippy is complaining about it.